runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java

```diff
@@ -160,6 +175,20 @@ public NettyClientRouter(String host, Integer port) {
         outstandingRequests = new ConcurrentHashMap<>();
         shutdown = true;
 
+        MetricRegistry metrics = CorfuRuntime.getMetrics();
+        String pfx = CorfuRuntime.getMpCR() + host + ":" + port.toString() + ".";
+        synchronized (metrics) {
+            if (!metrics.getNames().contains(pfx + "connected")) {
+                gaugeConnected = metrics.register(pfx + "connected", () -> connected_p ? 1 : 0);
+            }
+        }
+        timerConnect = metrics.timer(pfx + "connect");
+        timerSyncOp = metrics.timer(pfx + "sync-op");
+        counterConnectFailed = metrics.counter(pfx + "connect-failed");
+        counterSendDisconnected = metrics.counter(pfx + "send-disconnected");
```

Comment:
If the socket isn't connected at the time of the op (checked in `sendMessageAndGetCompletable()`) or if the socket is disconnected prior to receiving a reply (performed by `connectChannel()` via the `channel.closeFuture()` mechanism), then this counter will be incremented, once per per attempted or in-flight client op.